core/src/main/java/binnie/core/machines/storage/ComponentCompartmentInventory.java

```diff
@@ -55,14 +58,15 @@ public void sendGuiNBTToClient(final Map<String, NBTTagCompound> nbt) {
 			list.appendTag(nbt2);
 		}
 		final NBTTagCompound tag = new NBTTagCompound();
-		tag.setTag("tabs", list);
-		nbt.put("comp-tabs", tag);
+		tag.setTag(NBT_KEY_TABS, list);
+		nbt.put(NBT_KEY_COMP_TABS, tag);
 	}
 
 	@Override
-	public void receiveGuiNBTOnClient(EntityPlayer player, String name, NBTTagCompound nbt) {
-		if (name.equals("comp-tabs")) {
-			final NBTTagList tags = nbt.getTagList("tabs", 10);
+	public void receiveGuiNBTOnClient(final EntityPlayer player, final String name, final NBTTagCompound nbt) {
+		// TODO: RusTit: I think here should be ACTION_COMP_CHANGE_TAB key, need testing.
+		if (name.equals(NBT_KEY_COMP_TABS)) {
```

Comment:
To @Nedelosk @temp1011 
Need your advise.
As you can see, my commit do some refactoring.
Please take a look on methods: 
receiveGuiNBTOnClient and receiveGuiNBTOnServer
I think that in receiveGuiNBTOnClient code (original) `if (name.equals("comp-tabs")) {` is wrong.
Correct code should use `"comp-change-tab"` key (as in `receiveGuiNBTOnServer`).
It also used in `WindowCompartment.java` to provide action.

Reply:
That is the problem.
Most likely it is a bug and a string the constant is specified incorrectly.